src/main/java/org/jilt/internal/AbstractBuilderGenerator.java

```diff
@@ -333,13 +335,34 @@ protected final TypeVariableName mangleTypeParameter(TypeVariableName typeVariab
     private ParameterSpec setterParameterSpec(VariableElement attribute, TypeName parameterType) {
         ParameterSpec.Builder ret = ParameterSpec.builder(parameterType,
                 this.attributeSimpleName(attribute));
-        AnnotationMirror nullableAnnotation = this.firstAnnotationCalledNullable(attribute);
-        if (nullableAnnotation != null) {
-            ret.addAnnotation(AnnotationSpec.get(nullableAnnotation));
-        }
+        this.addAnnotationsToParam(ret, attribute.getAnnotationMirrors());
+        this.addAnnotationsToParam(ret, attribute.asType().getAnnotationMirrors());
         return ret.build();
     }
 
+    private void addAnnotationsToParam(ParameterSpec.Builder ret, List<? extends AnnotationMirror> annotationMirrors) {
+        for (AnnotationMirror annotation : annotationMirrors) {
+            if (this.isAnnotationAllowedInParameter(annotation)) {
+                ret.addAnnotation(AnnotationSpec.get(annotation));
+            }
+        }
+    }
+
+    public boolean isAnnotationAllowedInParameter(AnnotationMirror annotation) {
+        Target targetAnnotation = annotation.getAnnotationType().asElement().getAnnotation(Target.class);
+        if (targetAnnotation == null) {
+            return true;
+        }
+
+        for (ElementType elementType : targetAnnotation.value()) {
+            if (ElementType.TYPE_USE.equals(elementType) || ElementType.PARAMETER.equals(elementType)) {
```

Comment:
Let's flip the order in this check, `PARAMETER` is the more important one:

```suggestion
            if (ElementType.PARAMETER.equals(elementType) || ElementType.TYPE_USE.equals(elementType)) {
```